### ISSUE TITLE
[development branch] Compiler cache server infinite loop error

### ISSUE THREAD

**Issue by haxiomic** [HaxeFoundation/haxe#10283]
Using development branch haxe as a server via vscode, it randomly gets stuck infinite loops with errors like

```
PreprocessIcons.hx:39: characters 2-77 : Uncaught exception Stack overflow
/usr/local/lib/haxe/std/haxe/macro/Context.hx:1: character 1 : Called from here
/usr/local/lib/haxe/std/haxe/macro/Context.hx:51: characters 10-26 : Called from here
Assets.hx:15: characters 4-140 : Called from here
Assets.hx:19: characters 13-36 : Called from here
PreprocessIcons.hx:39: characters 2-77 : Called from here
/usr/local/lib/haxe/std/haxe/macro/Context.hx:51: characters 10-36 : Called from here
Assets.hx:15: characters 4-140 : Called from here
Assets.hx:19: characters 13-36 : Called from here
PreprocessIcons.hx:39: characters 2-77 : Called from here
/usr/local/lib/haxe/std/haxe/macro/Context.hx:51: characters 10-36 : Called from here
Assets.hx:15: characters 4-140 : Called from here
Assets.hx:19: characters 13-36 : Called from here
... repeating hundreds of times ...
```

Macro code is used in the project but the error happens when I’m not touching macro code and a language server restart resolves the issue

Reported to be happening for @nadako too

No minimal repro unfortunately but opened an issue to keep track. Keen to learn more if anyone has any additional information

**Comment by Simn** [comment]
I'm not sure what to do with this and how to investigate things further...

**Comment by haxiomic** [comment]
It's tricky since it's such a heisenbug without a clear repro – I think for now best we can do is maybe message / upvote here if it happens for you along with the haxe version so we can tell if it's a new thing / getting worse / getting better

I can send over the ProcessIcons.hx project if it's helpful, the bug is semi frequent, I maybe had it 5 times in a few days while developing

**Comment by back2dos** [comment]
A curiosity worth pointing out: the stack trace contains what looks to be `Context.error` multiple times (which shouldn't ever call Haxe code, unless I'm mistaken).

**Comment by Simn** [comment]
Not sure what to do here, might have to close unless more information is provided.

**Comment by kLabz** [comment]
Can try something if @haxiomic sends something to reproduce.
Also might have been fixed by #10933

**Comment by haxiomic** [comment]
I think it's fair to close this one as no repro

congratulations on 4.3! Amazing release!
